### PR TITLE
ADCM-2377 Speed up dev tests

### PR DESCRIPTION
**Pull request by Sealwing**
This PR is dedicated to speeding up "branch to develop" tests run.

To achieve it two things were done:
1. Some tests were marked as "full" because they take too much time or cover corner cases.
2. API tests were reworked to use "ADCM with dummy data" to skip the phase of pre-filling ADCM that takes about a minute for each test.

Also, the process of choosing which ADCM to use (clean or with dummy data) was reworked so it can be used in API tests the same way as it is used in functional and UI tests.

### CODE REVIEW

**Comment by dgusakov** [comment]
@Sealwing Pls add comprehensive description

File: tests/functional/test_adcm_upgrade.py [hunk 1]
```diff
@@ -121,7 +121,8 @@ def test_upgrade_adcm(
 
 
 @pytest.mark.parametrize("adcm_is_upgradable", [True], indirect=True)
-@pytest.mark.parametrize("image", old_adcm_images(), ids=repr, indirect=True)
+@pytest.mark.parametrize("image", old_adcm_images(), ids=repr)
+@pytest.mark.skip()
```
Comment (skhomuti) [inline tests/functional/test_adcm_upgrade.py:125]: remove it

Comment (Sealwing) [reply tests/functional/test_adcm_upgrade.py]: done

File: tests/functional/test_adcm_upgrade.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def _check_encryption(obj: Union[Cluster, Service]) -> None:
 
 
 @pytest.mark.parametrize("adcm_is_upgradable", [True], indirect=True)
-@pytest.mark.parametrize("image", old_adcm_images(), ids=repr, indirect=True)
+@pytest.mark.parametrize("image", old_adcm_images(), ids=repr)
```
Comment (dgusakov) [inline tests/functional/test_adcm_upgrade.py:105]: Why did you remove indirect here?

Comment (Sealwing) [reply tests/functional/test_adcm_upgrade.py]: returned it back

**Review by dgusakov — APPROVED** [review]

**Review by skhomuti — APPROVED** [review]
